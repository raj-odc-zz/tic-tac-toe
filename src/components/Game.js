/*
 * You do not need to separate components and containers.
 * Feel free to connect your state and components directly in here (or other components).
 */
import React from 'react';
import { connect } from 'react-redux';

import Board from './Board';

const Game = () => (
  <div>
    <header>TicTacToe</header>

    <div className="Game">
      <Board />
    </div>
  </div>
);

export default connect(
  // do be implemented
)(Game);
