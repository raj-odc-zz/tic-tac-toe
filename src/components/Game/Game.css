.Game {
  display: block;
  margin: 0 auto;

  height: var(--board-size);
  width: var(--board-size);
}

.Game > h2 {
  text-align: center;
}

.Game__User {
  width: 80%;
  max-width: 1200px;
  margin: 0 auto;
}

.Game__User * {
  box-sizing: border-box;
  max-width: 600px;
  margin: 0 auto;
}

.Game__User--field {
  display: flex;
  flex-wrap: wrap;
  text-align: left;
  padding: 8px;
  font-weight: 300;
  letter-spacing: .09em;
}

.Game__User--field > label {
  padding: 15px;
}

.Game__User--field input {
  padding: 15px;
}

.Game__User--submit {
  text-align: center;
}

.Game__User--submit button {
  margin-left: auto;
  padding: 8px 16px;
  background-color: #DCDCDC;
  text-transform: uppercase;
  letter-spacing: .09em;
  border-radius: 2px;
}

.displayErrors input:invalid {
  border-color: #f73772;
}