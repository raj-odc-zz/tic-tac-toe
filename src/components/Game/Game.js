import React, { Component } from 'react';
import { connect } from 'react-redux';

import Board from '../Board/Board';

import './Game.css';

import { createBoard, setPlayerName } from '../../store/actions';

class Game extends Component {
  constructor(props) {
    super(props);
    this.state = {
      displayErrors: false,
      player1: '',
      player2: '',
    };
  }

  handleChange = (e) => {
    this.setState({
      [e.target.name]: e.target.value,
    });
  }

  handleSubmit = (e) => {
    const { player1, player2 } = this.state;
    const { updatePlayers } = this.props;
    e.preventDefault();
    if (!e.target.checkValidity()) {
      this.setState({
        displayErrors: true,
      });
    } else {
      updatePlayers({ 1: player1, 2: player2 });
    }
  }

  renderForm = () => {
    const { displayErrors, player1, player2 } = this.state;
    return (
      <React.Fragment>
        <h2>Enter player names</h2>
        <form
          onSubmit={this.handleSubmit}
          noValidate
          className={displayErrors ? 'displayErrors' : ''}
        >
          <div className="Game__User--field">
            First Player Name
            <input
              type="text"
              id="player1"
              key="player1"
              name="player1"
              placeholder="Enter player name here"
              required
              value={player1}
              onChange={this.handleChange}
            />
          </div>
          <div className="Game__User--field">
            Second Player Name
            <input
              type="text"
              key="player2"
              name="player2"
              placeholder="Enter player name here"
              required
              value={player2}
              onChange={this.handleChange}
            />
          </div>
          <div className="Game__User--submit">
            <button type="submit">Submit</button>
          </div>
        </form>
      </React.Fragment>
    );
  }

  render() {
    const { players } = this.props;
    return (
      <div>
        <header>TicTacToe</header>
        <div className="Game">
          {
            players ? <Board /> : this.renderForm()
          }
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  activePlayer: state.activePlayer,
  players: state.players,
});

const mapDispatchToProps = dispatch => ({
  reInitateBoard: () => dispatch(createBoard()),
  updatePlayers: players => dispatch(setPlayerName(players)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Game);
