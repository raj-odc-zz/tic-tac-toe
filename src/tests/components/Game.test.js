import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { shallow, mount, render } from 'enzyme';

import configureStore from 'redux-mock-store';

import Game from '../../components/Game/Game.js';

const initialState = {
  board: [],
  activePlayer: null,
  winner: null,
  players: null,
};
const mockStore = configureStore();
const store = mockStore(initialState);

describe('<Game />', () => {
  let wrapper;

  beforeEach(() => {
     wrapper = shallow(<Game store={store}/>);
  });

  it('should have a `<form>` element', () => {
    wrapper = mount(<Game store={store}/>);
    expect(wrapper.find('form').length).toEqual(1);
  });

  it('should render the Game component', () => {
    expect(wrapper.length).toEqual(1)
  });

  it('check Prop matches with initialState', () => {
    expect(wrapper.prop('activePlayer')).toEqual(initialState.activePlayer)
 });

  it('renders a player1 name input', () => {
    wrapper = mount(<Game store={store}/>);
    expect(wrapper.find('[name="player1"]').length).toEqual(1)
  });

  it('should render to static HTML', function() {
    expect(render(<Game store={store}/>).text())
    .toEqual('TicTacToeEnter player namesFirst Player NameSecond Player NameSubmit');
  });
});
