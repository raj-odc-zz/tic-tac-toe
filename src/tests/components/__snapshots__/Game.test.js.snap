// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Game /> with initial state renders correctly with User Form 1`] = `
<div>
  <header>
    TicTacToe
  </header>
  <div
    className="Game"
  >
    <h2>
      Enter player names
    </h2>
    <form
      className=""
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="Game__User--field"
      >
        First Player Name
        <input
          id="player1"
          name="player1"
          onChange={[Function]}
          placeholder="Enter player name here"
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="Game__User--field"
      >
        Second Player Name
        <input
          name="player2"
          onChange={[Function]}
          placeholder="Enter player name here"
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="Game__User--submit"
      >
        <button
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</div>
`;
